src/drivers/rc_input/RCInput.cpp

```diff
@@ -308,21 +308,20 @@ void RCInput::Run()
 		adc_report_s adc;
 
 		if (_adc_sub.update(&adc)) {
-			const unsigned adc_chans = sizeof(adc.channel_id) / sizeof(adc.channel_id[0]);
+			if (adc.channel_id[ADC_RC_RSSI_CHANNEL] == ADC_RC_RSSI_CHANNEL) {
```

Comment:
This is not necessarily true. You need to find the channel.

Reply:
Does this means that array index is not the same as channel ID?

Reply:
Correct, that is why there was a loop around this before.